### ISSUE TITLE
PHP Fatal error:  Class 'Hm_Auth_IMAP' not found

### ISSUE THREAD

**Issue by gevatter** [jasonmunro/hm3#7]
I copied the sample ini file and the only thing i changed is `auth_type=IMAP`.

    PHP Fatal error:  Class 'Hm_Auth_IMAP' not found in /var/www/cypht/hm3-master/lib/session.php on line 118

I am aware that the comment says ` the imap module must be enabled` but it is present in the "modules" line at the end of the file. Am I missing something, is there something else what needs to be done? I din't find a hint what to do.

**Comment by jasonmunro** [comment]
Could be a bug, I don't use the IMAP authentication very often :) I will setup a test and see if I can reproduce this.

**Comment by jasonmunro** [comment]
Definitely a bug. Working on a fix now!

**Comment by jasonmunro** [comment]
Should be resolved, sorry about that! I moved the module set specific auth classes to the module code, but that creates some issues with not loading when we need them to. I will come up with a better fix, but in the meantime I moved the back to lib/ to resolve this issue. Thanks for the feedback again!

**Comment by gevatter** [comment]
No need to be sorry, thank you for your amazing quick answers :)

Unforunaliy I still get an error:

     Fatal error: Class 'Hm_IMAP' not found in /var/www/cypht/hm3-master/lib/auth.php on line 193

I cloned the whole repo in a clean envoronment to be sure.

**Comment by jasonmunro** [comment]
Well I did a pretty crappy job fixing that then, didn't I :) I had some local changes that partially masked this. Once I rewound that I was able to reproduce. It should be all fixed up now. I also noticed the default module list in the sample ini file is incorrect, so you will want to update that before trying again. https://github.com/jasonmunro/hm3/commit/d74206984238ad23d681d55c0b292019ea24a865. (be sure to rerun the config_gen.php script after updating your ini file). Thanks for the great feedback, I love getting this fixed up!
